src/Infrastructure.VS/IErrorListHelper.cs

```diff
@@ -0,0 +1,87 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS
+{
+    public interface IErrorListHelper
+    {
+        bool TryGetRuleIdFromSelectedRow(IErrorList errorList, out SonarCompositeRuleId ruleId);
```

Comment:
I just realized that I dont really need to pass in the error list everytime, the other classes do not need it anymore so I suppose it would make sense to just, pass it in during construction of this class and save it. (Only issue is im not quite sure how to do this as we are newing it up in the ctor of the other class.)